src/esp/nav/PathFinder.cpp

```diff
@@ -1165,6 +1174,56 @@ PathFinder::Impl::getTopDownView(const float pixelsPerMeter,
   return topdownMap;
 }
 
+const assets::MeshData::ptr PathFinder::Impl::getNavMeshData() {
+  if (meshData_ == nullptr && isLoaded()) {
+    meshData_ = assets::MeshData::create();
+    std::vector<esp::vec3f>& vbo = meshData_->vbo;
+    std::vector<uint32_t>& ibo = meshData_->ibo;
+
+    // Iterate over all tiles
+    for (int iTile = 0; iTile < navMesh_->getMaxTiles(); ++iTile) {
+      const dtMeshTile* tile =
+          const_cast<const dtNavMesh*>(navMesh_.get())->getTile(iTile);
+      if (!tile)
+        continue;
+
+      // Iterate over all polygons in a tile
+      for (int jPoly = 0; jPoly < tile->header->polyCount; ++jPoly) {
+        // Get the polygon reference from the tile and polygon id
+        dtPolyRef polyRef = navMesh_->encodePolyId(iTile, tile->salt, jPoly);
+        const dtPoly* poly = nullptr;
+        const dtMeshTile* tmp = nullptr;
+        navMesh_->getTileAndPolyByRefUnsafe(polyRef, &tmp, &poly);
+
+        CORRADE_INTERNAL_ASSERT(poly != nullptr);
+        CORRADE_INTERNAL_ASSERT(tmp != nullptr);
+
+        // Code to iterate over triangles from here:
```

Comment:
Let's add a `getPolygonTriangles` helper method or something.  This code is kinda weird/complicated and we should avoid duplicating it.

Reply:
Good idea. Makes the code look much cleaner and supports ease of future efforts.